staramr/tests/integration/detection/test_AMRDetectionPlasmid.py

```diff
@@ -0,0 +1,69 @@
+import logging
+import os
+import tempfile
+import unittest
+from os import path
+
+import pandas as pd
+from Bio import SeqIO
+
+from staramr.blast.BlastHandler import BlastHandler
+from staramr.blast.pointfinder.PointfinderBlastDatabase import PointfinderBlastDatabase
+from staramr.blast.resfinder.ResfinderBlastDatabase import ResfinderBlastDatabase
+from staramr.blast.plasmidfinder.PlasmidfinderBlastDatabase import PlasmidfinderBlastDatabase
+from staramr.databases.AMRDatabasesManager import AMRDatabasesManager
+from staramr.databases.resistance.pointfinder.ARGDrugTablePointfinder import ARGDrugTablePointfinder
+from staramr.databases.resistance.resfinder.ARGDrugTableResfinder import ARGDrugTableResfinder
+from staramr.detection.AMRDetection import AMRDetection
+from staramr.detection.AMRDetectionResistance import AMRDetectionResistance
+
+logger = logging.getLogger('AMRDetectionPlasmid')
+
+
+class AMRDetectionPlasmid(unittest.TestCase):
+
+    def setUp(self):
+        blast_databases_repositories = AMRDatabasesManager.create_default_manager().get_database_repos()
+        self.resfinder_dir = blast_databases_repositories.get_repo_dir(
+            'resfinder')
+        self.pointfinder_dir = blast_databases_repositories.get_repo_dir(
+            'pointfinder')
+        self.plasmidfinder_dir = blast_databases_repositories.get_repo_dir(
+            'plasmidfinder')
+
+        self.resfinder_database = ResfinderBlastDatabase(self.resfinder_dir)
+        self.resfinder_drug_table = ARGDrugTableResfinder()
+        self.pointfinder_drug_table = ARGDrugTablePointfinder()
+        self.plasmidfinder_database = PlasmidfinderBlastDatabase(
+            self.plasmidfinder_dir)
+        self.pointfinder_database = None
+        self.blast_out = tempfile.TemporaryDirectory()
+        self.blast_handler = BlastHandler(
+            {'resfinder': self.resfinder_database, 'pointfinder': self.pointfinder_database, 'plasmidfinder': self.plasmidfinder_database}, 2, self.blast_out.name)
+
+        self.outdir = tempfile.TemporaryDirectory()
+        self.amr_detection = AMRDetectionResistance(self.resfinder_database, self.resfinder_drug_table,
+                                                    self.blast_handler, self.pointfinder_drug_table,
+                                                    self.pointfinder_database, output_dir=self.outdir.name)
+
+        self.test_data_dir = path.join(path.dirname(__file__), '..', 'data')
+
+    def tearDown(self):
+        self.blast_out.cleanup()
+        self.outdir.cleanup()
+
+    def testPlasmidfinderNameSuccess(self):
```

Comment:
Awesome :+1: 

Could I get you to add some additional code at the end of this method?

So, if you look at this test (https://github.com/phac-nml/staramr/blob/master/staramr/tests/integration/detection/test_AMRDetection.py#L183) you will see the code:

```python
        hit_file = path.join(self.outdir.name, 'resfinder_beta-lactam-blaIMP-42-del-start.fsa')
        records = SeqIO.to_dict(SeqIO.parse(hit_file, 'fasta'))

        self.assertEqual(len(records), 1, 'Wrong number of hit records')

        expected_records = SeqIO.to_dict(SeqIO.parse(file, 'fasta'))
        self.assertEqual(expected_records['blaIMP-42_1_AB753456'].seq, records['blaIMP-42_1_AB753456'].seq,
"records don't match")
```

What this is verifying is that the file that gets written to the `out/hits/` directory after running staramr (containing the nucleotide data found by staramr matching to resistance genes/plasmids) is valid.

You should only need to change `resfinder_beta-lactam-blaIMP-42-del-start.fsa` to `plasmidfinder_test-plasmids-seq.fsa` (that is `plasmidfinder_[your test file name]`). And also change `blaIMP-42_1_AB753456` to the id used in the test file (https://github.com/phac-nml/staramr/blob/add_plasmidfinder_and_tests/staramr/tests/integration/data/test-plasmids-seq.fsa#L1).

What the code does is reads in the nucleotide sequence written in the `hits/` directory and compares it to the nucleotide sequence we used as input (they should be identical since we made no changes).

Reply:
Ah ok makes sense, already added in 36238aa